src/waveform/visualplayposition.cpp

```diff
@@ -59,14 +57,18 @@ void VisualPlayPosition::set(
 double VisualPlayPosition::calcOffsetAtNextVSync(
         VSyncThread* pVSyncThread, const VisualPlayPositionData& data) {
     if (data.m_audioBufferMicroS != 0.0) {
+        int refToVSync;
+        int syncIntervalTimeMicros;
 #ifdef MIXXX_USE_QML
-        Q_UNUSED(pVSyncThread);
-        const int refToVSync = 0;
-        const int syncIntervalTimeMicros = 0;
-#else
-        const int refToVSync = pVSyncThread->fromTimerToNextSyncMicros(data.m_referenceTime);
-        const int syncIntervalTimeMicros = pVSyncThread->getSyncIntervalTimeMicros();
+        if (!CmdlineArgs::Instance().isQml()) {
```

Comment:
This condition seems to be inverted. The waveforms are no longer smooth.

Reply:
Good catch! I have removed the `!` so the condition should be correct again. Are the waveforms smooth again for you?

Reply:
Confirmed, waveforms run smooth again!